### DISCUSSION TITLE
belongsToMany creates wrong sql for one relation, correct sql for the other relation

### DISCUSSION THREAD

**Discussion by nightcoding** [filamentphp/filament#10140] [General]
Hi there,

I am currently playing around with filament3 and creating an admin of one my productive applications. First of all: great work, I can not remember a time where I was so fast in producing outstanding results!

Now my problem: ich have three models: `Course`, `Location` and `Category`. Each `course` can have multiple `locations` and `categories`.
Therefore I have the following database tables (be aware: legacy application!):

* `courses`
* `locations`
* `categories`
* `course_location`
* `course_category`

I have succesuffly created the relations in the `course` model as `BelongsToMany`. Here are the definitions:

```
    public function locations(): BelongsToMany
    {
        return $this->belongsToMany(Location::class);
    }

    public function categories(): BelongsToMany
    {
        return $this->belongsToMany(Category::class);
    }
```

Now I wanted to add a multiple select in the edit action of the filament course resource, which I have done with the following code:

```
                Forms\Components\Select::make('locations')
                    ->multiple()
                    ->relationship(name: 'locations', titleAttribute: 'name')
                    ->searchable()
                    ->preload(),
```

and

```
                Forms\Components\Select::make('categories')
                    ->multiple()
                    ->relationship(name: 'categories', titleAttribute: 'name')
                    ->searchable()
                    ->preload(),
```

And this does not work! Especially for `categories`. If I comment out the select for the `categories`, everything works fine and great. I can select multiple records, delete records, etc. pp.

As soon as I comment in the `categories` select again, filament (or laravel) tells me the table `category_course` can not be found. Which is true, as there is no `category_course`. The table name is `course_category`.

But: the join table for `Location` is also called `course_location`. 

So why does it work for one relation (here `locations`) and not for the other (here `categories`)? Is this a bug or do I miss something?

Thanks!
Martin

**Comment by ashokssrkc** [comment]
I know this is very late for the reply but, It will be useful for someone. 

`public function categories(): BelongsToMany
    {
        return $this->belongsToMany(Categories::class, 'course_category', 'course_id', 'category_id');
    }`

Know it will be saved on separate table.

**Reply by nightcoding** [reply]
Hi ashokssrkc,

It's not late! It's a useful and good answer! 😃 I tried it today and it works like a charme! Thanks a lot, you're answer is a great help for me 👏🏻 

Best
Martin
